src/components/portal/portal.js

```diff
@@ -1,35 +1,63 @@
 import React from 'react';
-import ReactPortal from 'react-portal';
 import PropTypes from 'prop-types';
+import { createPortal } from 'react-dom';
+import Browser from '../../utils/helpers/browser';
 
 class Portal extends React.Component {
-  static propTypes = {
-    /**
-     * Determine if the portal is visible or not
-     *
-     * @property open
-     * @type {Boolean}
-    */
-    open: PropTypes.bool,
 
+  static propTypes = {
     /**
      * The content of the portal.
      *
      * @property children
      * @type {Node}
      */
-    children: PropTypes.node
+    children: PropTypes.node,
+    /**
+     * The DomNode 
+     *
+     * @property node
+     * @type {Any}
+     */
+    node: PropTypes.any
+  }
+
+  componentWillUnmount() {
+    if (this.defaultNode) {
+      Browser.getDocument().body.removeChild(this.defaultNode);
+    }
+    this.defaultNode = null;
+  }
+
+  getDefaultDiv() {
+    
+    if (!this.props.node && !this.defaultNode && Browser.isDomAvailable()) {
```

Comment:
would `Browser.isDomAvailable()` ever be `false` here seeing as you check on the first line of the `render` function that it is `true`?

Reply:
Similarly would `this.props.node` ever be `true`, seeing as you're testing it in the `render` too?

Reply:
Maybe you could clean up this confusion by getting rid of the ternary operator in the `render` function and having this function just return whatever the correct node is (whether it creates one, uses the memoized one, or uses the one from `props.node`).

Reply:
Yes I could feel there was something sloppy here. in the general confusion, I forgot to come back to it.